src/common/Block/Adminhtml/Form/Field/Attributes.php

```diff
@@ -2,56 +2,54 @@
 
 declare(strict_types=1);
 
-namespace Aligent\FredhopperIndexer\Block\Adminhtml\Form\Field;
+namespace Aligent\FredhopperCommon\Block\Adminhtml\Form\Field;
 
-use Magento\CatalogSearch\Model\Indexer\Fulltext\Action\DataProvider;
+use Aligent\FredhopperCommon\Model\Data\AttributeDataProvider;
 use Magento\Framework\Exception\LocalizedException;
 use Magento\Framework\View\Element\Context;
 use Magento\Framework\View\Element\Html\Select;
 
 class Attributes extends Select
 {
 
-    private DataProvider $dataProvider;
-
     /**
-     * Attributes constructor.
      * @param Context $context
-     * @param DataProvider $dataProvider
+     * @param AttributeDataProvider $dataProvider
      * @param array $data
      */
     public function __construct(
         Context $context,
-        DataProvider $dataProvider,
+        private readonly AttributeDataProvider $dataProvider,
         array $data = []
     ) {
         parent::__construct($context, $data);
-        $this->dataProvider = $dataProvider;
     }
 
     /**
+     * Sets the name of the input field
+     *
      * @param $value
      * @return $this
-     * @throws LocalizedException
      */
     public function setInputName($value): Select
     {
-        return $this->setName($value);
+        return $this->setData('name', $value);
     }
 
     /**
-     * @return string
+     * @inheritDoc
+     * @throws LocalizedException
      */
     public function _toHtml(): string
     {
         if (empty($this->getOptions())) {
-            $attributes = $this->dataProvider->getSearchableAttributes();
+            $attributes = $this->dataProvider->getIndexableAttributes();
             foreach ($attributes as $attributeCode => $attribute) {
                 // only interested in attribute codes, not ids
                 if (is_numeric($attributeCode)) {
                     continue;
                 }
-                $this->addOption($attributeCode, $attribute->getStoreLabel());
+                $this->addOption($attributeCode, implode(' - ', [$attributeCode, $attribute->getStoreLabel()]));
```

Comment:
Why the obfuscated method of string concat? It's less code to do it with `.` operators.

```
implode(' - ', [$attributeCode, $attribute->getStoreLabel()])
$attributeCode . ' - ' . $attribute->getStoreLabel();
```

Reply:
Good question. Not really sure why I did that :facepalm: